File: go/space-age/space_age.go
package space

type Planet  string

var planet_lengths = map[Planet]float64 {
  "Earth": 1.0, 
  "Mercury": 0.2408467,
  "Venus": 0.61519726,
  "Mars": 1.8808158,
  "Jupiter": 11.862615,
  "Saturn": 29.447498,
  "Uranus": 84.016846,
  "Neptune": 164.79132,
}

func Age(seconds float64, planet Planet) float64 {
    earth_years := seconds/31557600

    planet_years := earth_years/planet_lengths[planet]
    return planet_years
}

File: erlang/space-age/src/space_age.erl
-module(space_age).

-export([ageOn/2, test_version/0]).

ageOn(Planet, Seconds) -> Seconds / year(Planet) / 31557600.

year(earth) -> 1.0;
year(mercury) -> 0.2408467;
year(venus) -> 0.61519726;
year(mars) -> 1.8808158;
year(jupiter) -> 11.862615;
year(saturn) -> 29.447498;
year(uranus) -> 84.016846;
year(neptune) -> 164.79132.

test_version() -> 1.
